go-server-server/go/default.go

```diff
@@ -1162,6 +1162,133 @@ func ConfigVrouterVrfIdRoutesPatch(w http.ResponseWriter, r *http.Request) {
     }
 }
 
+func ConfigVrfVrfIdRoutesPatch(w http.ResponseWriter, r *http.Request) {
+    w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+    db := &conf_db_ops
+    vars := mux.Vars(r)
+    var rt_tb_key string
+    vrf_id_str := vars["vrf_id"]
+
+    if vrf_id_str != "default" {
+        // Only default is supported
+        WriteRequestError(w, http.StatusInternalServerError, "Internal service error", []string{}, "")
+        return
+    }
+
+    var attr []RouteModel
+
+    err := ReadJSONBody(w, r, &attr)
+    if err != nil {
+        // The error is already handled in this case
+        return
+    }
+
+    pt := swsscommon.NewTable(db.swss_db, STATIC_ROUTE_TB)
+    defer pt.Delete()
+
+    var rt_tb_name string
+    var failed []RouteModel
+
+    for _, r := range attr {
+
+        rt_tb_name = STATIC_ROUTE_TB
+        rt_tb_key = generateDBTableKey(db.separator, rt_tb_name, vrf_id_str, r.IPPrefix)
+
+        cur_route, err := GetKVs(db.db_num, rt_tb_key)
+        if err != nil {
+            r.Error_code = http.StatusInternalServerError
+            r.Error_msg = "Internal service error"
+            failed = append(failed, r)
+        }
+
+        if r.Cmd == "delete" {
+            if cur_route == nil {
+                r.Error_code = http.StatusNotFound
+                r.Error_msg = "Not found"
+                failed = append(failed, r)
+            } else {
+                pt.Del(generateDBTableKey(db.separator, vrf_id_str, r.IPPrefix), "DEL", "")
+            }
+        } else {
+            route_map := make(map[string]string)
+            if r.IfName == "" {
+                route_map["nexthop"] = r.NextHop
+            } else {
+                route_map["ifname"] = r.IfName
+                if r.NextHop != "" {
+                    route_map["nexthop"] = r.NextHop
+                }
+            }
+            if r.IfName == "null" {
+                route_map["blackhole"] = "true"
+            } else {
+                route_map["blackhole"] = "false"
```

Comment:
Wondering if `route_map["blackhole"]` needs to match the number of nexthops in `route_map["nexthop"]` (e.g., if we add `10.1.0.57,10.1.0.59`, should this be `false,false`)?

Reply:
As we don't support ECMP based blackhole, prefer to keep it simple for any specific route to be required to be blackholed.

Reply:
I agree that we do not need ECMP based blackhole. My point here is that for non-blackhole scenario, we are adding `false` as value for blackhole field. As we are adding one `false` regardless of the number of nexthops, we might have some trouble?

Reply:
I guess it should be fine. Other thing is to add the blackhole only when ifName is null and ignore the 'false' case. But then if its a route moved from null to non-null, we will have some issue with updating the blackhole attribute.